### PR TITLE
chore: initial teamsfx eslint package & add @author eslint rule

### PR DESCRIPTION
**Pull request by LongOddCode**

### CODE REVIEW

File: packages/eslint-plugin-teamsfx/lib/index.js [hunk 1]
```diff
@@ -0,0 +1,24 @@
+/**
```
Comment (wenytang-ms) [inline packages/eslint-plugin-teamsfx/lib/index.js:1]: should add MS copyright at head?

File: packages/eslint-plugin-teamsfx/lib/rules/jsdocAuthor.js [hunk 2]
```diff
@@ -0,0 +1,73 @@
+/**
```
Comment (wenytang-ms) [inline packages/eslint-plugin-teamsfx/lib/rules/jsdocAuthor.js:1]: Copyright here

File: packages/extra-shot-mocha/package.json [hunk 3]
```diff
@@ -38,6 +38,7 @@
     "eslint-plugin-import": "^2.25.2",
     "eslint-plugin-no-secrets": "^0.8.9",
     "eslint-plugin-prettier": "^4.0.0",
+    "@microsoft/eslint-plugin-teamsfx": "file:../eslint-plugin-teamsfx",
```
Comment (wenytang-ms) [inline packages/extra-shot-mocha/package.json:41]: Why not move eslint-plugin-teamsfx to extra-shot-mocha?
Or instead of version

**Review by chagong — DISMISSED** [review]

File: packages/extra-shot-mocha/package.json (hunk 3 above)
Comment (LongOddCode) [reply packages/extra-shot-mocha/package.json:41]: peer deps issue

**Review by wenytang-ms — DISMISSED** [review]

**Comment by wenytang-ms** [comment]
You may add this plugin to .github/script/lernaDeps.json to seperate release.
or let it publish together

**Review by wenytang-ms — APPROVED** [review]

**Review by wenytang-ms — APPROVED** [review]

**Review by yiqing-zhao — APPROVED** [review]

**Review by chagong — DISMISSED** [review]

**Review by jayzhang — APPROVED** [review]

**Review by tecton — DISMISSED** [review]

**Review by Alive-Fish — DISMISSED** [review]

**Review by kuojianlu — DISMISSED** [review]

**Review by HuihuiWu-Microsoft — APPROVED** [review]

**Review by yiqing-zhao — DISMISSED** [review]

**Review by wenytang-ms — DISMISSED** [review]

**Review by Alive-Fish — DISMISSED** [review]

**Review by tecton — DISMISSED** [review]

**Review by wenytang-ms — APPROVED** [review]

**Review by jayzhang — APPROVED** [review]

**Pull request merged**